src/homepage/howTo/custom-listeners.html

```diff
@@ -0,0 +1,217 @@
+                    </ul>
+
+                    <header class="major">
+                        <h2><b>What is Mailbox Listener</b></h2>
+                    </header>
+
+                    <p>
+                        Mailbox Listener is a component in James Mailbox System.
+                        Every time, if there is an action applied on a mailbox(adding, deleting), or an email(adding, deleting, updating flags...).
+                        Then an event represented for that action is generated and delivered to all Listeners that have registered before.
+                        After receiving events, listeners retrieve information from the events then execute their business
+                        (Indexing emails in ElasticSearch, updating quota of users, detecting spam emails...)
+                    </p>
+
+                    <p>
+                        There are two kinds of Listener registration:
+                    </p>
+                    <ul>
+                        <li>
+                            <b>Group Registration</b>: If a Listener is registered as the Group Registration, it will receive all of events fired by James Mailbox System.
+                            You can write your own custom Listener and configure James to register it as a Group Registration.
+                        </li>
+                        <li>
+                            <b>Key Registration</b>: If a Listener is registered as the Key Registration, it will receive only events it is interested in.
+                            At the moment, James doesn't support custom listeners Key Registration
+                        </li>
+                    </ul>
+
+                    <header class="major">
+                        <h2><b>Use the custom BigMessageListener in James</b></h2>
+                    </header>
+
+                    <p>
+                        Once you have a custom Listener, setup James with that Listener is very simple.
+                        In this example, we will use the
+                        <a href="https://github.com/apache/james-project/blob/master/examples/custom-mailets/src/main/java/org/apache/james/examples/custom/listeners/BigMessageListener.java">BigMessageListener</a>
+                        which will listen events fired after emails are appended to users mailboxes,
+                        then for each email added event, determine the size of added email by getting size information from the event.
+                        If the size of the email is greater than or equals 1 MB, then BigMessageListener will add a "BIG_MESSAGE" flag to that email
+                    </p>
+
+                    <header class="major">
+                        <h3><b>Starting James with BigMessageListener</b></h3>
+                    </header>
+
+                    <p>
+                        When you write a custom Listener, you have to compile it and place the compiled jar package inside "extensions-jars" directory
```

Comment:
Oh, really? I'm almost sure it only works with guice. If so, please have a prerequisite section to the how-to